src/en/authors-hook-debug.md

```diff
@@ -92,8 +92,10 @@ after the unit comes up, making it difficult to start a debug-hooks session in
 time to intercept them. If you're having difficulties, you can temporarily
 return an error code from your `install` hook (e.g. add an `exit 1` at the end
 of it), and start your session only when the unit reports an [error status
-](./authors-hook-errors.html). You should then run `juju resolved --retry` for
-the affected unit, and go back to the debug-hooks session to interact.
+](./authors-hook-errors.html). Juju version 2.0 and up will automatically retry
```

Comment:
When I develop charms with Juju 2.0, I usually don't wait around for the hook to retry though. How about something like:

> Hooks in error must be retried manually in Juju releases prior to 2.0, by issuing the command `juju resolved --retry unit-name\#` for the affected unit. Juju version 2.0 and up will automatically retry hooks in error periodically. However, `juju resolved --retry` may still be used to retry the hook immediately. After retrying, go back to the debug-hooks session to interact with the Juju environment.

I'm probably not fit to review `juju debug-hooks` in particular. I never use it; I just use print statements to debug things and use `charm build ...; juju upgrade-charm ...; juju resolved --retry ...` to try changes out iteratively.